File: internal/model/change_event.go
package model

import "time"

// ChangeEvent represents a CI/CD or infrastructure change event.
type ChangeEvent struct {
	ID          uint       `json:"id" gorm:"primaryKey"`
	Source      string     `json:"source" gorm:"size:50;not null"`
	ChangeType  string     `json:"change_type" gorm:"size:50;not null;default:deploy"`
	Service     string     `json:"service" gorm:"size:255;default:''"`
	Environment string     `json:"environment" gorm:"size:50;default:''"`
	CommitSHA   string     `json:"commit_sha" gorm:"size:64;default:''"`
	Author      string     `json:"author" gorm:"size:255;default:''"`
	Description string     `json:"description" gorm:"type:text"`
	RiskLevel   string     `json:"risk_level" gorm:"size:20;default:low"`
	Metadata    JSONLabels `json:"metadata" gorm:"type:json"`
	Timestamp   time.Time  `json:"timestamp"`
	CreatedAt   time.Time  `json:"created_at"`
	DeletedAt   *time.Time `json:"deleted_at,omitempty" gorm:"index"`
}

func (ChangeEvent) TableName() string { return "change_events" }

File: web/src/api/change-event.ts
import request from './request'
import type { ApiResponse, PageData } from '@/types'

export interface ChangeEvent {
  id: number
  source: string
  change_type: string
  service: string
  environment: string
  commit_sha: string
  author: string
  description: string
  risk_level: string
  metadata: Record<string, string>
  timestamp: string
  created_at: string
  updated_at: string
}

export interface IngestChangeEventRequest {
  source: string
  change_type?: string
  service: string
  environment: string
  commit_sha?: string
  author?: string
  description?: string
  risk_level?: string
  metadata?: Record<string, string>
  timestamp?: string
}

export const changeEventApi = {
  list: (params?: { page?: number; page_size?: number; service?: string; environment?: string; source?: string; incident_id?: number }) =>
    request.get<ApiResponse<PageData<ChangeEvent>>>('/change-events', { params }),
  get: (id: number) =>
    request.get<ApiResponse<ChangeEvent>>(`/change-events/${id}`),
  ingest: (data: IngestChangeEventRequest) =>
    request.post<ApiResponse<ChangeEvent>>('/change-events', data),
  delete: (id: number) =>
    request.delete<ApiResponse<null>>(`/change-events/${id}`),
}
